scripts/servermanager.sh

```diff
@@ -0,0 +1,167 @@
+    # Force a fresh install of all
+    pp --success "\n>>> Doing a fresh install of the gameserver...\n\n"
+    if [[ -n $WEBHOOK_ENABLED ]] && [[ $WEBHOOK_ENABLED == "true" ]]; then
+        send_webhook_notification "Installing server" "Server is beeing installed" "$WEBHOOK_INFO_COLOR"
+    fi
+    ${steamcmd_dir}/steamcmd.sh +force_install_dir "$GAME_ROOT" +login anonymous +app_update 2394010 validate +quit
+    pp --success ">>> Done installing the gameserver.\n"
+}
+
+function update_server() {
+    # Force an update and validation
+    if [[ -n $STEAMCMD_VALIDATE_FILES ]] && [[ $STEAMCMD_VALIDATE_FILES == "true" ]]; then
+        pp --success "\n>>> Doing an update and validate of the gameserver files...\n\n"
+        if [[ -n $WEBHOOK_ENABLED ]] && [[ $WEBHOOK_ENABLED == "true" ]]; then
+            send_webhook_notification "Updating server" "Server is beeing updated and validated" "$WEBHOOK_INFO_COLOR"
+        fi
+        ${steamcmd_dir}/steamcmd.sh +force_install_dir "$GAME_ROOT" +login anonymous +app_update 2394010 validate +quit
+        pp --success ">>> Done updating and validating the gameserver files.\n"
+
+    else
+        pp --success "\n>>> Doing an update of the gameserver files...\n\n"
+        if [[ -n $WEBHOOK_ENABLED ]] && [[ $WEBHOOK_ENABLED == "true" ]]; then
+            send_webhook_notification "Updating server" "Server is beeing updated" "$WEBHOOK_INFO_COLOR"
+        fi
+        ${steamcmd_dir}/steamcmd.sh +force_install_dir "$GAME_ROOT" +login anonymous +app_update 2394010 +quit
+        pp --success ">>> Done updating the gameserver files.\n"
+    fi
+}
+
+
+### Signal Handling (Handlers & Traps)
+
+# Handler for SIGTERM from docker stop command
+function termHandler() {
+    stopServer
+}
+
+function start_handlers() {
+    pp --success "\n>>> Starting handlers...\n"
+
+    pp --info "> Starting termination handler...\n"
+    # If SIGTERM is sent to the process, call termHandler function
+    trap 'kill ${!}; termHandler' SIGTERM
+
+    pp --info "> Termination handler started!\n"
+}
+
+
+### Main Function
+
+function start_main() {
+    pp --success "\n>>> Starting main thread <<<\n"
+
+    check_for_default_credentials
+    setup_crons
+
+    # Check if server is installed, if not try again
+    if [ ! -f "${GAME_ROOT}/PalServer.sh" ]; then
+        install_server
+    fi
+    if [ "${ALWAYS_UPDATE_ON_START}" == "true" ]; then
+        update_server
+    fi
+    start_server
+}
+
+
+### Server Manager Initialization
+
+# Server manager is running in a loop, so we can restart the server
+while true
+do
+    pp --success ">>> Starting server manager <<<\n"
+    start_handlers
+
+    # Start the server manager
+    start_main &
+
+    killpid="$!"
+    pp --info "\n>>> Server main thread started with pid ${killpid}\n"
+    wait ${killpid}
+    
+    # pp --warning ">>> Server stopped, checking for restart or backup restore\n"
+
+    # # If file 'restart' in server_restart exists, we do a server restart
```

Comment:
Why is this commented out?
Does it work?
Is it final?
Do i need to review it when its commented out?